File: Mathematics/sieve of Eratosthenes/python/sieve-of-eratosthenes.py
n = int(input())
prime = [1]*(n + 1)

p = 2
while p*p <= n:
    if prime[p] == 1:
        for i in range(p*2, n + 1, p):
            prime[i] = 0
    p+= 1

for i in range(2, n + 1):
    if prime[i] == 1:
        print("p =  " + str(i))

File: Mathematics/sieve of Eratosthenes/golang/Sieve-of-Eratosthenes.go
package main

import "fmt"

var MAXN int = 1000

func Sieve() {
  prime := make([]bool, MAXN)
  for i, _ := range prime {
    prime[i] = true
  }
  for i := 2; i < MAXN; i += 1{
    for j := i; i*j < MAXN; j += 1 {
      prime[i*j] = false;
    }
  }
  for i, flag := range prime {
    if (flag == true && i > 1) {
      fmt.Println(i)
    }
  }
}

//execution starts here

func main(){
  fmt.Println("This program prints all the prime numbers from 2 to MAXN")
  Sieve()
}
